src/wechaty-mixins/puppet-mixin.ts

```diff
@@ -26,9 +27,13 @@ const puppetMixin = <MixinBase extends WechatifyUserModuleMixin> (mixinBase: Mix
     /**
      * @protected
      */
-    _puppet?: PUPPET.impl.Puppet
+    // Huan(202111): developing Post
+    // _puppet?: PUPPET.impl.Puppet
+    _puppet?: PuppetPost  // FIXME: use `PUPPET.impl.Puppet`
```

Comment:
This is just for testing and developing, to make the vscode ide happy.

In the final code, the `_puppet` should be `PuppetImpl` or `PuppetInterface` from `wechaty-puppet` module.

Reply:
Then shall we modify this PR to be the final code?

Reply:
The current design of the Puppet API in this PR, is only for testing the Post data structure, and all the Puppet API in this draft PR is just for the demo/drafting purpose, it needs more time to review them before it can be sent to the wechaty-puppet repo as a new PR.

But yes, we can modify this PR to propose what the final code looks like, please feel free to go ahead and we can continue discussing it based on this PR.